tools/validate.py

```diff
@@ -11,75 +15,173 @@ def load_validator_from_url(uri_env_name):
+    validator = load_validator_from_url("EVAL_REQUEST_SCHEMA_URL")
+    request_error = validate(validator, body)
+
+    if request_error:
+        return {
+            "message": "Schema threw an error when validating the request body.",
+            "error_thrown": request_error,
+        }
+
+    return None
+
+
+def validate_preview_request(body):
+    """
     Function to return any errors in the request body based on its schema.
     ---
     If there are no issues with the request body, then None is returned. Otherwise, a
     JSON-encodable response containing the schema and errors will be returned. Each
     element in the list of errors is a dictionary containing the error message and the
     path to the rule in the schema that has thrown the error.
     """
-    request_error = validate(request_validator, body)
+    validator = load_validator_from_url("PREVIEW_REQUEST_SCHEMA_URL")
+    request_error = validate(validator, body)
 
     if request_error:
         return {
-            "message":
-            "Schema threw an error when validating the request body.",
-            "error_thrown": request_error
+            "message": "Schema threw an error when validating the request body.",
+            "error_thrown": request_error,
         }
 
     return None
 
 
 def validate_response(body):
-    """    
+    """
+    Function to return any errors in the response body based on its schema.
+    ---
+    If there are no issues with the response body, then None is returned. Otherwise, a
+    JSON-encodable response containing the schema and errors will be returned. Each
+    element in the list of errors is a dictionary containing the error message and the
+    path to the rule in the schema that has thrown the error.
+    """
+    validator = load_validator_from_url("RESPONSE_SCHEMA_URL")
+    response_error = validate(validator, body)
+
+    if response_error:
+        return {
+            "message": "Schema threw an error when validating the response body.",
+            "error_thrown": response_error,
+            "raw_response_body": body,
+        }
+
+    return None
+
+
+def validate_eval_response(body):
+    """
+    Function to return any errors in the response body based on its schema.
+    ---
+    If there are no issues with the response body, then None is returned. Otherwise, a
+    JSON-encodable response containing the schema and errors will be returned. Each
+    element in the list of errors is a dictionary containing the error message and the
+    path to the rule in the schema that has thrown the error.
+    """
+    validator = load_validator_from_url("EVAL_RESPONSE_SCHEMA_URL")
+    response_error = validate(validator, body)
+
+    if response_error:
+        return {
+            "message": "Schema threw an error when validating the response body.",
+            "error_thrown": response_error,
+            "raw_response_body": body,
+        }
+
+    return None
+
+
+def validate_healthcheck_response(body):
```

Comment:
In hindsight, consider that all these validation functions do the same thing:

- load a validator from a URL
- run validate
- return a validation error if any, else None

You could reduce duplication by having a more general validation function which accepts a URL and a validation error function (like `request_validation_error` proposed above). Something like:

```python

def validate(validator_url, validation_error_builder):
     def _validate(body):
          validator = load_validator_from_url(validator_url)
          response_error = validate(validator, body)
          return validation_error_builder(response_error) if response_error else None
     return _validate
```

And then perhaps:

```
validate_request = validate("REQUEST_SCHEMA_URL", request_validation_error)
```

where `validate_request`, and `request_validation_error` come from the code I suggested in my comment above (lines 39-55).